app/models/event_type.rb

```diff
@@ -0,0 +1,76 @@
+class EventType
+  attr_accessor :type_id
+
+  delegate(
+    :lookup_by_id,
+    :online_event_id,
+    :school_or_university_event_id,
+    :train_to_teach_event_id,
+    :question_time_event_id,
+    to: :class,
+  )
+
+  class << self
+    def school_or_university_event_id
+      lookup_by_name("School or University event")
+    end
+
+    def train_to_teach_event_id
+      lookup_by_name("Train to Teach event")
+    end
+
+    def online_event_id
+      lookup_by_name("Online event")
+    end
+
+    def question_time_event_id
+      lookup_by_name("Question Time")
+    end
+
+    def lookup_by_name(name)
+      GetIntoTeachingApiClient::Constants::EVENT_TYPES.fetch(name)
+    end
+
+    def lookup_by_names(*names)
+      GetIntoTeachingApiClient::Constants::EVENT_TYPES.fetch_values(*names)
+    end
+
+    def lookup_by_id(id)
+      GetIntoTeachingApiClient::Constants::EVENT_TYPES.invert.fetch(id)
+    end
+
+    def lookup_by_ids(*ids)
+      GetIntoTeachingApiClient::Constants::EVENT_TYPES.invert.fetch_values(*ids)
+    end
+  end
+
+  def initialize(event)
+    @event   = event
+    @type_id = event.type_id
+  end
+
+  def type
+    lookup_by_id(type_id)
+  end
+
+  # this name might be confusing...
+  def online_event?
```

Comment:
I wonder if its just worth calling out that this is the type of the event, so `online_event_type?`... though you'd probably need to suffix the others to be consistent

Reply:
Yeah - it crossed my mind. Perhaps now we're referring to them as 'Online Q&A events' everywhere `#online_qa_event?` would work?